packages/flutter/lib/src/material/switch_theme.dart

```diff
@@ -0,0 +1,228 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui' show lerpDouble;
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/rendering.dart';
+import 'package:flutter/widgets.dart';
+
+import 'material_state.dart';
+import 'theme.dart';
+import 'theme_data.dart';
+
+/// Defines default property values for descendant [Switch] widgets.
+///
+/// Descendant widgets obtain the current [SwitchThemeData] object using
+/// `SwitchTheme.of(context)`. Instances of [SwitchThemeData] can be customized
+/// with [SwitchThemeData.copyWith].
+///
+/// Typically a [SwitchThemeData] is specified as part of the overall [Theme]
+/// with [ThemeData.switchTheme].
+///
+/// All [SwitchThemeData] properties are `null` by default. When null, the
+/// [Switch] will use the values from [ThemeData] if they exist, otherwise it
+/// will provide its own defaults based on the overall [Theme]'s colorScheme.
+/// See the individual [Switch] properties for details.
+///
+/// See also:
+///
+///  * [ThemeData], which describes the overall theme information for the
+///    application.
+@immutable
+class SwitchThemeData with Diagnosticable {
+  /// Creates a theme that can be used for [ThemeData.switchTheme].
+  const SwitchThemeData({
+    this.thumbColor,
+    this.trackColor,
+    this.materialTapTargetSize,
+    this.mouseCursor,
+    this.focusColor,
+    this.hoverColor,
+    this.splashRadius,
+  });
+
+  /// {@macro flutter.material.switch.thumbColor}
+  ///
+  /// If specified, overrides the default value of [Switch.thumbColor].
+  final MaterialStateProperty<Color?>? thumbColor;
```

Comment:
Should these properties be the ones to be themeable instead of OR alongside `thumbColor` and `trackColor`?

```dart
    this.activeColor,
    this.activeTrackColor,
    this.inactiveThumbColor,
    this.inactiveTrackColor,
```

Reply:
The properties `thumbColor` and `trackColor` is to be used instead of these. MaterialStateProperty is the "future proof" way of handling the colors in different shapes etc, and this theme would follow the pattern of how the new Buttons work.